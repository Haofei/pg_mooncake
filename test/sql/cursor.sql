CREATE TABLE t (a int);
INSERT INTO t VALUES (1);

BEGIN;
DECLARE c SCROLL CURSOR FOR SELECT a FROM t;
FETCH NEXT FROM c;
FETCH NEXT FROM c;
FETCH PRIOR FROM c;
COMMIT;

DROP TABLE t;

CREATE TABLE t (a int) USING columnstore;
INSERT INTO t VALUES (1);

BEGIN;
DECLARE c SCROLL CURSOR FOR SELECT a FROM t;
FETCH NEXT FROM c;
FETCH NEXT FROM c;
FETCH PRIOR FROM c;
COMMIT;

DROP TABLE t;
