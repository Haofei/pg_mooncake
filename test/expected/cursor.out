CREATE TABLE t (a int);
INSERT INTO t VALUES (1);
BEGIN;
DECLARE c SCROLL CURSOR FOR SELECT a FROM t;
FETCH NEXT FROM c;
 a 
---
 1
(1 row)

FETCH NEXT FROM c;
 a 
---
(0 rows)

FETCH PRIOR FROM c;
 a 
---
 1
(1 row)

COMMIT;
DROP TABLE t;
CREATE TABLE t (a int) USING columnstore;
INSERT INTO t VALUES (1);
BEGIN;
DECLARE c SCROLL CURSOR FOR SELECT a FROM t;
FETCH NEXT FROM c;
 a 
---
 1
(1 row)

FETCH NEXT FROM c;
 a 
---
(0 rows)

FETCH PRIOR FROM c;
 a 
---
 1
(1 row)

COMMIT;
DROP TABLE t;
